### PR TITLE
Fix sherpa morphology fitting script

### PR DESCRIPTION
**Pull request by adonath**
This fixes the Sherpa morphology fitting script and adds a test, that runs the script on the `poisson_stats_image` test dataset.

### CODE REVIEW

**Comment by adonath** [comment]
@cdeil This currently fails on Travis-Ci.

File: gammapy/scripts/tests/test_sherpa_like.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import json
+import os
+
+from astropy.io import fits
+
+from ...datasets import load_poisson_stats_image
+from ..sherpa_like import sherpa_image_like
+from numpy.testing.utils import assert_allclose
+
+
```
Comment (cdeil) [inline gammapy/scripts/tests/test_sherpa_like.py]: You have to mark the test as using the optional Sherpa dependency:
https://travis-ci.org/gammapy/gammapy/jobs/77172790#L1154
http://astropy.readthedocs.org/en/latest/development/testguide.html#tests-requiring-optional-dependencies

And please add a changelog entry for this.

**Comment by adonath** [comment]
@cdeil Thanks, I've fixed the test. It still fails with an error, that I can't reproduce locally. Maybe it's related to the sherpa version (I use 'latest')?

**Comment by cdeil** [comment]
@olaurino or @DougBurke – We're starting to use Sherpa in Gammapy, and on travis-ci we're getting this AttributeError for `sherpa.astro.io` that @adonath doesn't see locally: https://travis-ci.org/gammapy/gammapy/jobs/77185047#L1419

As shown in https://github.com/gammapy/gammapy/blob/master/.travis.yml#L116 we're using `conda install -c https://conda.binstar.org/cxc sherpa` and for @adonath the test works locally, also using conda.

Any ideas what the issue is or how to fix it?

File: gammapy/scripts/sherpa_like.py [hunk 2]
```diff
@@ -93,11 +93,17 @@ def sherpa_image_like(counts,
...
+    #sherpa.astro.ui.covar()  # Computes symmetric errors (fast)
     # conf() # Computes asymmetric errors (slow)
     # image_fit() # Shows data, model, residuals in ds9
     log.info('Writing {}'.format(outfile))
     write_all(outfile)
+
+    # Save model image
+    sherpa.astro.ui.set_par('background.ampl', 0)
+    sherpa.astro.ui.notice2d()
+    logging.info('Writing model.fits')
```
Comment (cdeil) [inline gammapy/scripts/sherpa_like.py]: Please follow [this guideline](https://gammapy.readthedocs.org/en/latest/development/howto.html#generating-log-messages) to generate log messages.

**Comment by cdeil** [comment]
I left one minor inline comment and the travis-ci Sherpa AttributeError has to be fixed (I don't know how).
Otherwise, this looks good to me and is ready to merge.

**Comment by DougBurke** [comment]
It looks like `sherpa.astro.ui.utils` doesn't directly import `sherpa.astro.io`, so it's not clear when it gets into scope. As to why this is seen on Travis and not locally is a tad strange. I don't suppose you can do something really stupid like add a `import sherpa.astro.io` statement to the test and see if that "fixes" it (as we don't have a more-up-to-date version of sherpa on conda yet I'm looking for simple-ish fixes to get you going while we understand the issue more ;-)?

**Comment by olaurino** [comment]
@cdeil The issue itself is odd, especially if you don't experience it locally. I'll try and have a look asap.

In the meantime, while it might not fix your issue, you may want to switch to https://conda.binstar.org/sherpa as this is the official channel with the latest release binaries.

If you need a development cut with the latest master code I can probably arrange that, and eventually my plan is to have such development cuts to be automatically generated.

**Comment by olaurino** [comment]
The problem seems to be that the sherpa version you are trying to use (and, for that matter, the latest official release), uses pyfits for I/O. We did add support for astropy (and fall back to pyfits), and that will work if you use the master branch:

see https://travis-ci.org/olaurino/gammapy/jobs/77343964

There are several options to fix this. I am not sure which one is preferable for you: you could install sherpa from sources, but that's probably overkill, especially since you are using the legacy Travis infrastructure so you cannot cache the build. You could install pyfits via conda to let sherpa have its I/O backend. We do ship a pyfits conda binary in the same channel as sherpa.

I could indeed provide you with a development conda package for Linux64. While I cannot commit to a date, I could try to do that by the end of the week.

I can't double check I am not missing anything obvious, but I believe the following line in your .travis.yml works "by accident":
````
if $OPTIONAL_DEPS && [[ $SETUP_CMD != egg_info ]] && [[ $TRAVIS_PYTHON_VERSION == 2.7 ]] && [[ $NUMPY_VERSION == 1.8 ]];
then $CONDA_INSTALL -c https://conda.binstar.org/cxc sherpa; fi
````

I could not find the `OPTIONAL_DEPS` env variable in the script... it should probably be `INSTALL_OPTIONAL`. Even in that case, if you set the variable to "false" the condition `if $INSTALL_OPTIONAL` would evaluate, I believe, to True. Same if the variable is undefined. Since this condition always evaluates to True (again, unless I am missing something), the other conditions actually determine whether sherpa gets installed, i.e. when Python 2.7 and numpy 1.8 are used.

By the way, Sherpa should support numpy 1.9, if you can trick conda into it, e.g.: `conda install -y sherpa && conda install -y numpy=1.9`. We will update the conda package's metadata to reflect this for the next release.

**Comment by cdeil** [comment]
My preferred solution here would be that @adonath just skips the test for now and we merge this.

Then when @olaurino can produce a development conda package for Sherpa in the coming weeks, we switch the Gammapy travis-ci to use that and activate the test.

**Comment by olaurino** [comment]
I uploaded a sherpa conda package built from the latest master branch (commit sherpa/sherpa/@2d013d129a37375b9866ea5efefd40b325a52ba1) to the following channel:
https://conda.anaconda.org/cxc/channel/dev

Please let me know if you have any issues.

**Comment by adonath** [comment]
@olaurino Thanks for looking into this!
@cdeil I've still marked the test to be skipped, as I won't have time to work on this during the coming days and next week.

**Comment by cdeil** [comment]
I'll change the travis-ci build to use the dev version of Sherpa and activate the test in a new PR later today.

This looks good to me and travis-ci passed ... merging now.

@adonath – Thanks for finishing this up and enjoy your vacation!

**Comment by cdeil** [comment]
FYI: In #340 I've updated the Gammapy travis-ci setup.
Before it was a modified version of the Astropy afiiliated `package-template`.
Now it is a modified version of the Astropy core repo travis-ci setup.

The test implemented in this PR using Sherpa is working.

**Pull request merged**